### PR TITLE
SASS-6481: Load prior data for Income journey

### PR DESCRIPTION
**Pull request by lukastymo**

### CODE REVIEW

File: app/services/journeyAnswers/IncomeAnswersService.scala [hunk 1]
```diff
@@ -25,17 +25,24 @@ import models.domain.ApiResultT
...
 trait IncomeAnswersService {
+  def getAnswers(businessId: BusinessId, taxYear: TaxYear, mtditid: Mtditid): ApiResultT[Option[IncomeJourneyAnswers]]
   def saveAnswers(businessId: BusinessId, taxYear: TaxYear, mtditid: Mtditid, answers: IncomeJourneyAnswers): ApiResultT[Unit]
 }
 
 @Singleton
-class IncomeAnswersServiceImpl @Inject() (repository: MongoJourneyAnswersRepository)(implicit ec: ExecutionContext) extends IncomeAnswersService {
+class IncomeAnswersServiceImpl @Inject() (repository: JourneyAnswersRepository)(implicit ec: ExecutionContext) extends IncomeAnswersService {
+
+  def getAnswers(businessId: BusinessId, taxYear: TaxYear, mtditid: Mtditid): ApiResultT[Option[IncomeJourneyAnswers]] =
```
Comment (harry-ritchie) [inline app/services/journeyAnswers/IncomeAnswersService.scala:41]: Can we add a UT for this?

File: app/controllers/package.scala [hunk 2]
```diff
@@ -14,28 +14,34 @@
...
 import play.api.libs.json.Format.GenericFormat
-import play.api.libs.json.{JsError, JsResult, JsSuccess, Json, Reads}
-import play.api.mvc.Results.{BadRequest, InternalServerError}
+import play.api.libs.json._
+import play.api.mvc.Results._
 import play.api.mvc.{AnyContent, Result}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 package object controllers {
```
Comment (harry-ritchie) [inline app/controllers/package.scala:28]: Should we add UTs for package objects?

File: app/services/journeyAnswers/IncomeAnswersService.scala (hunk 1 above)
Comment (lukastymo) [reply app/services/journeyAnswers/IncomeAnswersService.scala:41]: added

File: app/controllers/package.scala (hunk 2 above)
Comment (lukastymo) [reply app/controllers/package.scala:28]: yes once it stabilizes.

right now we will test it by using it in controllers - writing test controller will test some of the helpers

File: test/services/journeyAnswers/IncomeAnswersServiceImplSpec.scala [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  "saveAnswers" should {
+    "save data in the repository" in new TestCase() {
+      service
+        .saveAnswers(JourneyContextWithNino(currTaxYear, businessId, mtditid, nino), incomeJourneyAnswersData)
+        .value
+        .futureValue shouldBe ().asRight
+    }
+  }
+}
```
Comment (harry-ritchie) [inline test/services/journeyAnswers/IncomeAnswersServiceImplSpec.scala:72]: There's already a test for this, it is under `IncomeAnswersServiceSpec`, could we bring it in here? Sorry my bad, probably should had named the file `IncomeAnswersServiceImplSpec`

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**